src/test/crimson/gtest_seastar.cc

```diff
@@ -0,0 +1,72 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "gtest_seastar.h"
+
+seastar_gtest_env_t seastar_test_suite_t::seastar_env;
+
+seastar_gtest_env_t::seastar_gtest_env_t() :
+  begin_fd{seastar::file_desc::eventfd(0, 0)} {}
+
+void seastar_gtest_env_t::init(int _argc, char **_argv)
+{
+  argc = _argc;
+  argv = new char *[argc];
+  for (int i = 0; i < argc; ++i) argv[i] = strdup(_argv[i]);
+
+  thread = std::thread([this] { reactor(); });
+  eventfd_t result = 0;
+  if (int r = ::eventfd_read(begin_fd.get(), &result); r < 0) {
+    std::cerr << "unable to eventfd_read():" << errno << std::endl;
+    throw std::runtime_error("Cannot start seastar");
+  }
+}
+
+void seastar_gtest_env_t::stop()
+{
+  run([this] {
+    on_end->write_side().signal(1);
+    return seastar::now();
+  });
+  thread.join();
+}
+
+seastar_gtest_env_t::~seastar_gtest_env_t()
+{
+  if (argv) {
```

Comment:
I don't see any real reason to.

Reply:
as we can assume that `init()` is always called, hence `argv` should already be set at this moment.

Reply:
Sure, I just figured it was easier to let it be more general in case at some point in the future it were possible for the env to be destructed prior to init.  I'll change it.